### PR TITLE
git-imerge 1.2.0

### PR DESCRIPTION
**Pull request by mochadwi**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/git-imerge.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class GitImerge < Formula
   desc "Incremental merge for git"
   homepage "https://github.com/mhagger/git-imerge"
-  url "https://github.com/mhagger/git-imerge/archive/v1.1.0.tar.gz"
+  url "https://github.com/mhagger/git-imerge/archive/v1.2.0.tar.gz"
   sha256 "62692f43591cc7d861689c60b68c55d7b10c7a201c1026096a7efc771df2ca28"
   license "GPL-2.0"
```
Comment (equal-l2) [inline Formula/git-imerge.rb:6]:
```suggestion
  license "GPL-2.0-or-later"
```
(guessed from https://github.com/mhagger/git-imerge/blob/0c1c05f631b2b42ae7eded0ea266a29b7871629f/gitimerge.py#L10)

File: Formula/git-imerge.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class GitImerge < Formula
   desc "Incremental merge for git"
   homepage "https://github.com/mhagger/git-imerge"
-  url "https://github.com/mhagger/git-imerge/archive/v1.1.0.tar.gz"
+  url "https://github.com/mhagger/git-imerge/archive/v1.2.0.tar.gz"
   sha256 "62692f43591cc7d861689c60b68c55d7b10c7a201c1026096a7efc771df2ca28"
```
Comment (Rylan12) [inline Formula/git-imerge.rb:5]: Thanks for the PR, @mochadwi!

You need to update the `sha256` line as well. It looks like the line should be:

```suggestion
  sha256 "cd80a817bc8dd4b5642f83610237e855fc0a23f5c11978d6f7d021cfb0e87bb2"
```

Additionally, please squash and rename your commit to `git-imerge 1.2.0` (the same as the name of this PR).

**Review by Rylan12 — CHANGES_REQUESTED** [review]

**Comment by mochadwi** [comment]
Is it also okay if I marked this PR as Hacktoberfest contribution? cc @Rylan12 or @equal-l2

File: Formula/git-imerge.rb (hunk 2 above)
Comment (mochadwi) [reply Formula/git-imerge.rb:5]: Well noted with thanks!

File: Formula/git-imerge.rb (hunk 1 above)
Comment (mochadwi) [reply Formula/git-imerge.rb:6]: Agreed, will modified this

**Comment by mochadwi** [comment]
I'm opening this PR due to inactivity in: https://github.com/Homebrew/homebrew-core/pull/61471 should I continue on #61471 ?

**Comment by Rylan12** [comment]
@chenrui333 any progress on #61471? If not, it's fine to continue here.

**Comment by mochadwi** [comment]
> @chenrui333 any progress on #61471? If not, it's fine to continue here.

I've squash my commit

**Comment by chenrui333** [comment]
looks like the build is still failing? cc @mochadwi 
```
==> brew install --verbose --build-bottle git-imerge
==> FAILED
==> Downloading https://github.com/mhagger/git-imerge/archive/v1.2.0.tar.gz
Already downloaded: /Users/brew/Library/Caches/Homebrew/downloads/462d0e1db00104012afc13fc63c1d5d471e20845dddc08a3ea8601c7053444e7--git-imerge-1.2.0.tar.gz
==> Verifying 462d0e1db00104012afc13fc63c1d5d471e20845dddc08a3ea8601c7053444e7--git-imerge-1.2.0.tar.gz checksum
/usr/bin/sandbox-exec -f /private/tmp/homebrew20201006-62995-ybbo21.sb nice ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/build.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/git-imerge.rb --build-bottle --verbose
tar xof /Users/brew/Library/Caches/Homebrew/downloads/462d0e1db00104012afc13fc63c1d5d471e20845dddc08a3ea8601c7053444e7--git-imerge-1.2.0.tar.gz -C /private/tmp/d20201006-62996-1ae5gse
cp -pR /private/tmp/d20201006-62996-1ae5gse/git-imerge-1.2.0/. /private/tmp/git-imerge-20201006-62996-kd16kr/git-imerge-1.2.0
chmod -Rf +w /private/tmp/d20201006-62996-1ae5gse
==> make install DESTDIR=/usr/local/Cellar/git-imerge/1.2.0 PREFIX=
make: *** No rule to make target `install'.  Stop.
```

**Comment by mochadwi** [comment]
> looks like the build is still failing? cc @mochadwi
> 
> ```
> ==> brew install --verbose --build-bottle git-imerge
> ==> FAILED
> ==> Downloading https://github.com/mhagger/git-imerge/archive/v1.2.0.tar.gz
> Already downloaded: /Users/brew/Library/Caches/Homebrew/downloads/462d0e1db00104012afc13fc63c1d5d471e20845dddc08a3ea8601c7053444e7--git-imerge-1.2.0.tar.gz
> ==> Verifying 462d0e1db00104012afc13fc63c1d5d471e20845dddc08a3ea8601c7053444e7--git-imerge-1.2.0.tar.gz checksum
> /usr/bin/sandbox-exec -f /private/tmp/homebrew20201006-62995-ybbo21.sb nice ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/build.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/git-imerge.rb --build-bottle --verbose
> tar xof /Users/brew/Library/Caches/Homebrew/downloads/462d0e1db00104012afc13fc63c1d5d471e20845dddc08a3ea8601c7053444e7--git-imerge-1.2.0.tar.gz -C /private/tmp/d20201006-62996-1ae5gse
> cp -pR /private/tmp/d20201006-62996-1ae5gse/git-imerge-1.2.0/. /private/tmp/git-imerge-20201006-62996-kd16kr/git-imerge-1.2.0
> chmod -Rf +w /private/tmp/d20201006-62996-1ae5gse
> ==> make install DESTDIR=/usr/local/Cellar/git-imerge/1.2.0 PREFIX=
> make: *** No rule to make target `install'.  Stop.
> ```

Is this a problem with `make` executable?

**Comment by SMillerDev** [comment]
That error looks like there is no `make install`

**Comment by mochadwi** [comment]
> ```
> make: *** No rule to make target `install'.  Stop.
> ```

Looks like the library author has some improvement in installation process, will modify this later

**Comment by yous** [comment]
`git-imerge` switched to use setup.py instead of Makefile: https://github.com/mhagger/git-imerge/pull/143

**Comment by mochadwi** [comment]
> `git-imerge` switched to use setup.py instead of Makefile: [mhagger/git-imerge#143](https://github.com/mhagger/git-imerge/pull/143)

That's right. I've edit it to use `pip` for `def install`. Is it okay?

**Comment by yous** [comment]
I think we need to do `python3 setup.py install`. By looking around other formulae,
``` rb
def install
  system Formula["python@3.9"].opt_bin/"python3", "setup.py", "install",
         "--prefix=#{prefix}",
         "--single-version-externally-managed",
         "--record=installed.txt"
end
```
Also `depends_on "python@3.9"` should be added.

**Comment by Rylan12** [comment]
You'll likely want to look into using `virtualenv_install_with_resources` (there are lots of formulae that use this you can look at for reference)

If that doesn't work you'll probably need to use something like:

```ruby
system Formula["python@3.8"].opt_bin/"python3", *Language::Python.setup_install_args(libexec)
```

Off the top of my head, I think you can use the [`urh`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/urh.rb) formula as reference.

I'd also recommend switching to using the PyPI URL (in this case, https://files.pythonhosted.org/packages/be/f6/ea97fb920d7c3469e4817cfbf9202db98b4a4cdf71d8740e274af57d728c/git-imerge-1.2.0.tar.gz). That way, `brew bump-formula-pr` will automatically add `resource` blocks if they become necessary in the future. This is not a deal-breaker though.

**Comment by mochadwi** [comment]
It fails when try to installing in my end: https://github.com/mhagger/git-imerge/issues/153

Any ideas to support this `bash_completions` in macOS?

**Comment by Rylan12** [comment]
You can install completions to the `bash_completion` or `zsh_completion` path. I don't know an example off the top of my head but if you search for `bash_completion` you should get lots of results.

**Comment by mochadwi** [comment]
> You can install completions to the `bash_completion` or `zsh_completion` path. I don't know an example off the top of my head but if you search for `bash_completion` you should get lots of results.

```

 depends_on "bash-completion"
 def install
  // install git-imerge
 end
```

also doesn't works, any ideas to configure bash completion inside `Formula/git-imerge.rb`?

**Comment by mochadwi** [comment]
came accross this approach: https://discourse.brew.sh/t/conditional-depends-on-on-formulae/1943/4

does it still relevant in the latest homebrew version using `bash_completion.install`?

**Comment by Rylan12** [comment]
Yes, `bash_completion.install` is the way to go. You don't need to depend on the `bash-completion` formula to use it. The `bash_completion.install` line goes in the `install` method (usually toward the bottom but I don't think that's a rule).

One example is the `velero` formula but there are countless more available for reference in Homebrew/core:

https://github.com/Homebrew/homebrew-core/blob/b80fdc60f4c4e254e2e5180061977005dc263086/Formula/velero.rb#L28-L34

**Comment by mochadwi** [comment]
https://github.com/mhagger/git-imerge/pull/176 the author now skip the `bash_completion` on ZSH shell

**Comment by chenrui333** [comment]
Close in favor of #66058

**Pull request closed without merging**